pkg/resourcekeeper/dispatch.go

```diff
@@ -118,29 +118,10 @@ func (h *resourceKeeper) record(ctx context.Context, manifests []*unstructured.U
 }
 
 func (h *resourceKeeper) dispatch(ctx context.Context, manifests []*unstructured.Unstructured) error {
-	var errs []error
-	var l sync.Mutex
-	var wg sync.WaitGroup
-
-	ch := make(chan struct{}, MaxDispatchConcurrent)
 	applyOpts := []apply.ApplyOption{apply.MustBeControlledByApp(h.app), apply.NotUpdateRenderHashEqual()}
-
-	for i := 0; i < len(manifests); i++ {
-		ch <- struct{}{}
-		wg.Add(1)
-		go func(index int) {
-			defer wg.Done()
-			manifest := manifests[index]
-			applyCtx := multicluster.ContextWithClusterName(ctx, oam.GetCluster(manifest))
-			err := h.applicator.Apply(applyCtx, manifest, applyOpts...)
-			if err != nil {
-				l.Lock()
-				errs = append(errs, err)
-				l.Unlock()
-			}
-			<-ch
-		}(i)
-	}
-	wg.Wait()
-	return kerrors.NewAggregate(errs)
+	errs := parallel.ParExec(func(manifest *unstructured.Unstructured) error {
```

Comment:
```suggestion
	errs := parallel.Run(func(manifest *unstructured.Unstructured) error {
```

We can omit the duplicate part of function name with the package name